cluster-autoscaler/core/static_autoscaler_test.go

```diff
@@ -2396,6 +2412,108 @@ func TestFilterOutYoungPods(t *testing.T) {
 	}
 }
 
+func TestStaticAutoscalerRunOnceInvokesScaleDownStatusProcessorNoCandidates(t *testing.T) {
+	options := config.AutoscalingOptions{
+		NodeGroupDefaults: config.NodeGroupAutoscalingOptions{
+			ScaleDownUnneededTime:         time.Minute,
+			ScaleDownUnreadyTime:          time.Minute,
+			ScaleDownUtilizationThreshold: 0.5,
+			MaxNodeProvisionTime:          10 * time.Second,
+		},
+		EstimatorName:    estimator.BinpackingEstimatorName,
+		ScaleDownEnabled: true,
+		MaxNodesTotal:    10,
+		MaxCoresTotal:    10,
+		MaxMemoryTotal:   100000,
+	}
+	now := time.Now()
+	n1 := BuildTestNode("n1", 1000, 1000)
+	SetNodeReadyState(n1, true, now)
+	p1 := BuildTestPod("p1", 800, 800, WithNodeName("n1"))
+
+	testCases := map[string]struct {
+		pods           []*apiv1.Pod
+		nodes          []*apiv1.Node
+		expectedStatus *status.ScaleDownStatus
+	}{
+		"Status processor is called with no scaledown candidates, utilized": {
+			pods:  []*apiv1.Pod{p1},
+			nodes: []*apiv1.Node{n1},
+			expectedStatus: &status.ScaleDownStatus{
+				Result:          status.ScaleDownNoUnneeded,
+				ScaledDownNodes: []*status.ScaleDownNode{},
+				UnremovableNodes: []*status.UnremovableNode{
+					{
+						Node:        n1,
+						BlockingPod: nil,
+						Reason:      simulator.NotUnderutilized,
+					},
+				},
+				RemovedNodeGroups:     nil,
+				NodeDeleteResults:     map[string]status.NodeDeleteResult{},
```

Comment:
We have to have something inside the results here, otherwise we're not verifying that they're passed correctly at all. I think your code has a bug right now (see the other comment), but the test isn't failing because this is empty.

Reply:
You're right, it was gathered in the wrong place, I'll conduct another pass of manual testing

Reply:
Made a run on a patched commit, status processor reports periodically with or without scaledown candidates

Reply:
It looks like still none of the test cases check if any actual results are reported? Or am I missing something here? If you run this test on the previous version which retrieved the results in the wrong place - does it start failing?

Reply:
If you are talking about NodeDeletionResults object itself it would be quite hard to get this part tested in a reliable way because those are being populated when async drain/deletion completed. Talking about whether it fails if I move that - due to the fact that it's empty in all tests - it's not failing

Reply:
Yeah, I specifically meant for the test to check if the results are reported, otherwise it wouldn't have caught the previous bug. The async part is why I though it might not be viable to implement such a test. Looking at the test code, it might not be that hard though. Wouldn't it be enough to make `nodeDeletionTracker` configurable through `autoscalerSetupConfig`, and then just fake-report the deletions directly through the tracker from the test code?

Reply:
Got your point, I've made a kind of mocked node deletion tracker to simulate the behavior